### DISCUSSION TITLE
[HLS] Improving HLS with Shared Accelerators:A Retrospective

### DISCUSSION THREAD

**Discussion by pmokri87** [cucapra/latte21#31] [Papers]
Author(s): Parnian Mokri, Mark Hempstead
Paper PDF: https://capra.cs.cornell.edu/latte21/paper/7.pdf
Talk Recording: https://youtu.be/uoRFMrrOqU4

**Comment by rachitnigam** [comment]
@pmokri87 Your YouTube talk is marked as "content for kids" which means we can't add it to our playlist. Can you update the setting for the video and mark it not for kids? Please ping me (by saying @rachitnigam) when you're done.

**Comment by KFAFSP** [comment]
We had previously considered the reuse opportunities for much simpler, more homgeneous systems, but found one problem: the gain could be vastly improved if we modified the input code of each user, ensuring that there would be more overlap. This happens at the expense of different other factors. Of course, that is to be expected, but we had wondered how we could use the reuse mapping to iteratively (or even predictively) make this decision. Have you made any observations in that regard?

**Reply by pmokri87** [reply]
That's a great point. Yes, I agree that in general memory system for accelerators can mak or break the optimizations in accelerators.  In reconfAST, we assumed that accelerators are blocking and don't share input data, which would make sense considering the accelerators ar from different domains. However, in our next paper we are looking in to similarities between input data and communication between accelerators as general using modified graph representations as well.

**Reply by pmokri87** [reply]
Just to clarify, are you talking about the user's code or the input data and how the function is read?

**Reply by KFAFSP** [reply]
I must have missed a notification here, I apologize. Still, to clarify, I am talking about the actual code. We are desigining custom tooling that also generates the accelerator implementation, so we do have full control over that aspect. The input data, on the other hand, can't be controlled and is only known to have some more general properties.

**Comment by soronpo** [comment]
If there is a shareable interface between accelerators, did you consider whether or not its better to use your proposed solution over partial configuration? Is there a need to quickly switch between accelerators?

**Reply by pmokri87** [reply]
Great questions. In ReconfAST the accelerators were blocking and there wasn't a need to switch between them. The system we imagined for our system was closer to the accelerator store paper. ReconfAST was a proof of concept on whether accelerators from different domains share any similarities. And if so, how can we design a tool that finds those similarities from the source code. In other words, it can be a tool to find these similarities and sits on top of the HLS tools.
